### ISSUE TITLE
Possible bug passing threshold values to pct()

### ISSUE THREAD

**Issue by ryancbriggs** [naoki-egami/evalid#3]
Consider the following case:

```
library(evalid)
p_values <- c(0.05, 0.06, 0.1, 0.02)

pct(p_values)
```
Which produces the following table:
```
  threshold p_value h_num
1         1    0.08     4
2         2    0.15     1
3         3    0.15     2
4         4    0.15     3
```

However, when I specify the thresholds directly I sometimes get different results. For example:
```
> pct(p_values, threshold = 1)
[1] 0.08
> pct(p_values, threshold = 2)
[1] 0.15
> pct(p_values, threshold = 3)
[1] 0.12
> pct(p_values, threshold = 4)
[1] 0.1
```
I believe this is incorrect behaviour.
(I'm using evalid version 1.0.0.)

**Comment by naoki-egami** [comment]
@ryancbriggs Thank you so much for using our package and reporting this point. 

It is a correct behavior (while I definitely admit that we did not document this point well, which we will!).  

When we just call `pct` without specifying `threshold`, the function checks all thresholds at the same time, and thus, we have to take care of the concerns of multiple testing. In particular, the function tests all four hypotheses at the same time: (1) whether there is at least one statistically significant effect, (2) whether there are at least two statistically significant effects, (3) whether there are at least three statistically significant effects, (4) whether there are at least four statistically significant effects. Therefore, the function has to properly account for this in the resulting p-values. This is what you see as the output of `pct` (without specifying any threshold). 

When we call `pct` with a specific `threshold`, we are only testing one hypothesis. For example, when we set `threshold = 2`, we only test whether there are at least two effects that are statistically significant. Therefore, there is no need to do multiple testing corrections. This is the reason why you see the p-value when you specify the threshold is always smaller than when you test all thresholds together (without specifying thresholds). 

In practice, when researchers have a specific hypothesis (e.g., they want to check whether the effects are positive in more than half of the sites they tested), they can explicitly specify the threshold they pick. If researchers want to simply test sign generalization, we usually recommend testing all thresholds together to reduce the risk of p-hacking and improve transparency. We provide more discussions in the relevant section of the paper (file:///Users/naokiegami/Documents/naoki-egami.github.io/paper/external_full.pdf)! 

Thank you again for your question. I hope this explanation makes sense. Please let us know if you have any further questions!

**Comment by ryancbriggs** [comment]
I see. This makes sense.

I ended up in a weird place because I looked at the tabular output but then was hoping to call the function with the threshold argument in order to make it easier to drop the number in the text in a clean way (e.g. in a quarto document). When the numbers were then different I got confused and worried. This use case may not be common, and of course I can just save the table and then pull numbers out of that.

I appreciate the quick response and the public goods you're providing here.
